plugins/woocommerce-admin/client/products/hooks/use-variation-orders.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { useFormContext } from '@woocommerce/components';
+import type { ProductVariation } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+
+const KEY_SEPARATOR = ':';
+
+function getVariationKey( variation: ProductVariation ) {
+	return `${ variation.id }${ KEY_SEPARATOR }${ variation.menu_order }`;
+}
+
+function getVariationId( { key }: JSX.Element ) {
+	return typeof key === 'string'
+		? Number.parseInt( key.split( KEY_SEPARATOR )[ 0 ], 10 )
+		: 0;
+}
+
+function getVariationOrder( { key }: JSX.Element ) {
+	return typeof key === 'string'
+		? Number.parseInt( key.split( KEY_SEPARATOR )[ 1 ], 10 )
+		: Number.MAX_SAFE_INTEGER;
+}
+
+function sort(
+	variations: ProductVariation[],
+	currentPage: number,
+	{ variationOrders }: ProductVariationOrders
+) {
+	if ( ! variationOrders || ! variationOrders[ currentPage ] )
+		return variations;
+
+	const currentPageOrders = variationOrders[ currentPage ];
+
+	return [ ...variations ].sort( ( a, b ) => {
+		if ( ! currentPageOrders[ a.id ] || ! currentPageOrders[ b.id ] )
+			return 0;
+		return currentPageOrders[ a.id ] - currentPageOrders[ b.id ];
+	} );
+}
+
+export default function useVariationOrders( {
```

Comment:
Could we update all cases of `variationOrders` to `variationsOrder`?

This avoids some ambiguity around "orders" placed by customers.

Reply:
done -> [2a5fa9d](https://github.com/woocommerce/woocommerce/pull/36109/commits/2a5fa9d09a8491f7aefc9fa608e4c81d93eaf556)